bundles/framework/timeseries/instance.js

```diff
@@ -80,161 +41,78 @@ Oskari.clazz.define("Oskari.mapframework.bundle.timeseries.TimeseriesToolBundleI
             }
             me.started = true;
 
-            var conf = me.conf || {},
-                sandboxName = (conf ? conf.sandbox : null) || 'sandbox',
-                sandbox = Oskari.getSandbox(sandboxName);
-            me.setSandbox(sandbox);
-            this.localization = Oskari.getLocalization(this.getName());
+            var sandboxName = (me.conf ? me.conf.sandbox : null) || 'sandbox';
+            var sandbox = me._sandbox = Oskari.getSandbox(sandboxName);
 
-            var mapModule = sandbox.findRegisteredModuleInstance('MainMapModule');
-            var plugin = Oskari.clazz.create('Oskari.mapframework.bundle.timeseries.TimeseriesAnimationPlugin', mapModule);
-            mapModule.registerPlugin(plugin);
-            mapModule.startPlugin(plugin);
-            this._plugin = plugin;
+            me._timeseriesService = Oskari.clazz.create('Oskari.mapframework.bundle.timeseries.TimeseriesService');
+            sandbox.registerService(me._timeseriesService);
 
-            sandbox.register(me);
-            for (p in me.eventHandlers) {
-                if (me.eventHandlers.hasOwnProperty(p)) {
-                    sandbox.registerForEventByName(me, p);
-                }
-            }
-            //because of timing issues, the eventhandlers may or may not be initialised by the time a time series layer is added ->
-            //so check this out manually the first time around and set the eventhandlers only after.
-            me._checkIfTimeseriesLayersExist();
-        },
-        /**
-        * @method init
-        * implements Module protocol init method - initializes request handlers
-        */
-        init: function(){
-            var me = this;
-            var sandbox = me.getSandbox();
-            var mapModule = sandbox.findRegisteredModuleInstance('MainMapModule');
-            var locale = this.getLocalization('timeseriesplayback');
-            var playback = Oskari.clazz.create('Oskari.mapframework.bundle.timeseries.TimeseriesControl', this, me.conf, locale, mapModule, sandbox);
-            me._modules.playback = playback;
-        },
-        /**
-         * @method initTimeSeries
-         * inits the timeline ui with values of a layer supporting times
-         */
-        initTimeseries: function(layer) {
-            var me = this,
-                times = layer.getAttributes().times,
-                layerId = layer.getId();
+            me._timeseriesLayerService = Oskari.clazz.create('Oskari.mapframework.bundle.timeseries.TimeseriesLayerService', sandbox, me._timeseriesService);
+            sandbox.registerService(me._timeseriesLayerService);
+            me._timeseriesLayerService.registerLayerType('wms', function (layerId) {
```

Comment:
This should probably be in the wmslayerplugin, but since wms has "hardcoded" support in maplayerservice it's fine in here too.

Reply:
Yes, the thinking was that since 'Oskari.mapframework.bundle.timeseries.WMSAnimator' class is part of timeseries bundle, we should register it for use here. Another option would be to move WMSAnimator to be a part of the general WMS layer functionality, but then the code would be loaded even if it cannot be used (requires timeseries bundle). I think it's better this way.

Reply:
True, lets keep it like this.